lib/puppet/type/http.rb

```diff
@@ -1,17 +1,53 @@
-Puppet::Type.newtype :http, :is_capability => true do
+Puppet::Type.newtype :http do
+  @doc = "A type for testing http connections"
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
   newparam(:name, :is_namevar => true) do
-    desc 'The name of the resource'
+    desc 'The unique name of the resource'
+  end
+
+  newparam(:uri) do
+    desc 'The URL to test'
+    defaultto "http://127.0.0.1/"
   end
 
-  newparam(:ip) do
-    desc 'IP address of the HTTP service'
+  newparam(:timeout) do
+    desc 'Time before timing out the resource (seconds)'
+    defaultto 60
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
+      Float(value)
+    end
   end
 
-  newparam(:port) do
-    desc 'Port of the HTTP service'
+  newparam(:ping_interval) do
+    desc 'Time to sleep before attempting to try a connection again (second)'
+    defaultto 1
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
+      Float(value)
+    end
   end
 
-  newparam(:host) do
-    desc 'Hostname of the HTTP service'
+  newparam(:status_codes) do
+    desc 'An array of HTTP status codes that will return success. For example
+          [200, 302] will match for OK and redirection responses.'
+    defaultto [200]
```

Comment:
Oh interesting, nagios check_http uses a response string matching system also, not status codes:

https://www.monitoring-plugins.org/doc/man/check_http.html